File: remove-duplicates-sorted.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int numSize = nums.size();
        if (numSize == 0) {
            return 0;
        }
        int shift = 0;
        int prev = nums[0];
        for (int i = 0; i < numSize; i++) {
            if (i != 0 && nums[i-shift] == prev) {
                nums.erase(nums.begin()+(i-shift));
                shift++;
            }
            prev = nums[i-shift];
        }
        return nums.size();
    }
};

File: remove-duplicates-sorted.java
class Solution {
    public int removeDuplicates(int[] nums) {
        Set<Integer> map = new HashSet<Integer>();
        int shift = 0;
        for (int i = 0; i < nums.length; i++) {
            if (map.contains(nums[i + shift])) {
                for (int j = (i + shift); j < nums.length - 1; j++) {
                    nums[j] = nums[j+1];
                }
                shift--;
            } else {
                map.add(nums[i + shift]);
            }
        } 
        return nums.length + shift;
    }
}
